xarray/backends/api.py

```diff
@@ -136,17 +136,10 @@ def _get_engine_from_magic_number(filename_or_obj):
         engine = "scipy"
     elif magic_number.startswith(b"\211HDF\r\n\032\n"):
         engine = "h5netcdf"
-        if isinstance(filename_or_obj, bytes):
-            raise ValueError(
-                "can't open netCDF4/HDF5 as bytes "
-                "try passing a path or file-like object"
-            )
     else:
-        if isinstance(filename_or_obj, bytes) and len(filename_or_obj) > 80:
-            filename_or_obj = filename_or_obj[:80] + b"..."
         raise ValueError(
-            "{} is not a valid netCDF file "
-            "did you mean to pass a string for a path instead?".format(filename_or_obj)
+            "{} is not the signature of any supported file format "
```

Comment:
```suggestion
            f"{magic_number} is not the signature of any supported file format "
```

Reply:
Can we user f-strings in xarray! Lovely!